### DISCUSSION TITLE
Define a Home Office colour palette

### DISCUSSION THREAD

**Discussion by LaurencePiercy** [UKHomeOffice/design-system#355] [Ideas]
We are designing numerous services which use colour. 

We need colour to distinguish between different categories and actions. This helps our users to understand these categories or actions. 

To enable the design community to establish a consistent and accessible colour palette, please add examples below.

We are looking for examples which demonstrate the **range** and **uses** of colours in Home Office services. 

If you are unable to screenshot your service, please add examples of the colours with a non-disclosing narrative description to help us understand the design rationale for colour use.

**Comment by LaurencePiercy** [comment]
https://github.com/UKHomeOffice/design-system/issues/291

**Comment by guermanc** [comment]
Discussion on colour should hopefully help up address #351

**Comment by LaurencePiercy** [comment]
Just consolidating a little:

- Background colour
- The colour of features (like radios and text boxes). These are currently white, background is grey.
- Tab colours (contrast with background and other features)
- Alert box colours (contrast with background and other features)
- Error colours (contrast with background and other features)
- Links (contrast with coloured components, like tab and alert)

**Reply by isobelseacombe** [reply]
I would add buttons to this colour review. We need secondary buttons that have the right level of contrast on the Home Office grey background (see https://github.com/UKHomeOffice/design-system/issues/291)

**Reply by isobelseacombe** [reply]
- Colour contrast of hint text on the grey background

**Comment by danibergho** [comment]
Entity Search (DSA) Colour usage examples
- Colour used in data tables, to help users search through and identify information they wish to further investigate
- Components with colour used include tags, labelling and error messages 
- We have seen in research that use of colour has helped our users to navigate through large quantities of information to find key characteristics or important features they are looking for

Examples:
1) Presenting characteristics on information cards using tags, example below as shown on design system: https://design-system.service.gov.uk/components/tag/

![Group 30](https://user-images.githubusercontent.com/95030057/164703926-2b9138b6-088b-438d-8b7c-53cd66f05717.png)
We chose to use these tags as it allowed us to seperate out the characteristics from the rest of the information, and highlight the information to users. 
.

2) Error messages within drop down 
<img width="327" alt="Screenshot 2022-04-22 at 12 17 12" src="https://user-images.githubusercontent.com/95030057/164704521-f1ae418e-402a-4e56-ad3d-c16126d42200.png">
Grey box used to seperate nested content, and error message therefore shown on the grey background in red. Red chosen to highlight the error that the user needs to investigate. In this example, blue is also used for the hyperlink button to fix this error.

.

3) Presenting severity levels of harm types:
<img width="171" alt="Screenshot 2022-04-22 at 12 15 43" src="https://user-images.githubusercontent.com/95030057/164704260-3042858f-7227-4701-accc-75bc66a28329.png">

Colours used: red #d4351c, orange #f47738, yellow #ffdd00. 
Colours chosen as research showed users understood these to represent severity levels and easily understood red to show higher level of harm, so high priority to investigate. 
Text labels were also used for each harm level to allow wider range of users who may not be able to understand colour labels to make use of them.

**Reply by LaurencePiercy** [reply]
Just to confirm, background colour for the service is white?

**Reply by danibergho** [reply]
Yes background is white for the page, but there is some drop down content contained within a grey box

**Comment by elliot-butler-ho** [comment]
A couple of component designs with improved accessible colours exist in our [styles and components figma file](https://www.figma.com/file/y90OmHLuhtNuCUOc7jzQlj/Home-Office-design-library-(Roboto)?node-id=4%3A88), including the secondary button #291  and labels / tags #43 . Although looks like these tag / labels designs were removed from the design system itself?

**Reply by LaurencePiercy** [reply]
Yes, they were removed from our design system because they were superseded by the gov.uk tags (as per Dani's example above). I think tags are back in scope for this review, because of the contrast issues we have against a grey background. (which is specific to our HO context)

**Comment by elliot-butler-ho** [comment]
In Manage Asylum Claim, I've also been looking at a tag colour to replace the existing grey from the GOV.UK design system https://design-system.service.gov.uk/components/tag/. We have two tag states (where a document is either submitted or not submitted). To replace the grey, I opted for a white background with 1px black border, which is based on one of the tag designs in Home Office figma file.

<img width="1029" alt="image" src="https://user-images.githubusercontent.com/80390605/164706668-9d7356e0-0877-4787-91b9-a61df0250214.png">

**Comment by LaurencePiercy** [comment]
https://github.com/UKHomeOffice/design-system/issues/43

**Comment by Richard592** [comment]
Laurence,

There is a related discussion to colour use on the Alerts component that Ounie, Ananda and I have contributed here: #358  
Essentially the tint colours fail accessibility when used in conjunction with the HO grey background and when links are presented within them. 

Generally for me colour is 'less is more' for all good accessibility reasons and should only be supplementing the design presentation and not the only means of communicating information. There is also a consistency element between GOV.UK and HO palettes that needs to be maintained where practical.

There was a fair amount of colour usage within the Callisto project before I left it, so hopefully that team will contribute some useful insights too.

**Comment by mikieet6** [comment]
Hi there, these are the colours we've been using on the ASL project. We use them on the tag components to highlight the status of various things in lists. The badges might contain different text based on the context, but the colours are used to indicate these themes:

- Positive: Things that are completed with a positive outcome
- Neutral: Things that are still in progress/incomplete 
- Negative: Things that are resolved with a negative outcome
- Changed: Things that have been changed since the previous submission
- Comment: Things that have had new comments added since the previous submission

<img width="124" alt="Screenshot 2022-06-09 at 09 39 18" src="https://user-images.githubusercontent.com/7107033/172804733-11f2e8bf-ac04-4651-a59a-88af177b2396.png">

**Comment by willrichards** [comment]
This is the current state of tags in the Central Operations Platform. At present only the top three (dark-grey, white, dark-blue) are planned to be used. We need to review all other tags now that we have changed from the GOV.UK design system to the Home Office design system.
 
<img width="1033" alt="COP-tags" src="https://user-images.githubusercontent.com/606378/173061794-0824b350-8809-4f75-b3ac-dca059183605.png">

**Comment by guermanc** [comment]
## Accessibility of Tags

### Colour

- Colour can be used on tags, but only if the text also supports the meaning conveyed by that colour. You cannot change the text of a tag and rely on colour to designate a category or severity, for example. 
 
However, if the tags do not require a user to accurately perceive a specific colour, you may use colour as a category and change the text within it, as long as:

- the colour of the tag has a contrast ratio of 3:1 against all other tag colours, as this counts as an alternative visual distinction
- the colour of the tag has a contrast ratio of 3:1 against the adjacent page background colour
- the colour of the text within the tag has a contrast ratio of 4.5:1 or 3:1 for large scale text against the tag colour
- the meaning of the tag colour can be programmatically determined, that is, available for assistive technologies such as screen readers

### Sizing

- users should be able to resize text up to 200% without any problems, such as truncated text or overlap with other content
- users should be able to magnify the page up to 400% and tags will reflow correctly and not add horizontal scrolling
- no loss of content or functionality happens when a user changes line height, letter or word spacing

### Position and styling

- the visual presentation of tags matches the programmatic implementation, so that someone using a screen readers or removing CSS encounters tags alongside whatever content they are supposed to be tagging

### Additional considerations
Things that are not WCAG failures but should be considered:

- when considering [progressive enhancement](https://www.gov.uk/service-manual/technology/using-progressive-enhancement), tags should still make sense with the CSS switched off
- some screen readers may interpret all capital text as being an acronym, and read out each individual letter
- ensure that you use [clear and understandable content](https://www.w3.org/TR/coga-usable/#objective-3-use-clear-and-understandable-content-0)

**Comment by isobelseacombe** [comment]
Work in progress in the design system working group:
- Developing content to explain the design decisions for the grey background     
- Reviewing the consistency and accessibility of colours on published styles and components on the design system 

To do:
- Develop a colour palette for Home Office Design system that works on the grey background and aligns with the GOV.UK colour palette
- Update existing components with the agreed colour palette 
- Update Figma design system file with the agreed colour palette

**Comment by tomyems** [comment]
I'd like to propose changing the hint text grey from 505a5f to 4e505f. This will increase the contrast with the page background from 6.26 (505a5f) to 7.05 (4e505f), and therefore meet WCAG 2.1 AAA.

Current:
![image](https://user-images.githubusercontent.com/93599300/207322913-d8e935b2-c276-4b6c-9aeb-5854dc081907.png)

Proposed:
![image](https://user-images.githubusercontent.com/93599300/207322846-4fe5e4d0-0f35-4460-bf15-b410d7ec8bd1.png)

I'd also like to propose changing the error message red from d4351c to a1151c. This will increase the contrast with the page background from 4.30 (d4351c) to 7.04 (aa1151c), and therefore meet WCAG 2.1 AAA.

Current:
![image](https://user-images.githubusercontent.com/93599300/207322693-f3386d93-437f-4fbf-a674-a9f77f5cf25b.png)

Proposed:
![image](https://user-images.githubusercontent.com/93599300/207322567-874941e9-807d-4452-aedd-24565d2f4c22.png)

**Reply by tomyems** [reply]
We could change the page background colour. However we have good evidence that the light grey background makes it easier for users to read. In particular those who have to see the same screen all day. See https://design.homeoffice.gov.uk/components?name=Page.

The proposed red does have purple tones, leading to it becoming maroon. There is some concern that this may reduce the effect of it's meaning.

The red A70000, which has no purple tones, is the lightest red with sufficient contrast for AAA (it's 7.03).

![image](https://user-images.githubusercontent.com/93599300/208059188-546810e0-f8af-470c-b3e1-d4ec1747d1be.png)

**Reply by Richard592** [reply]
Hey Tom, I think the A70000 is a good compromise removing the purple tones from the previous option. Works for me. Needs testing with service users when we next get a chance.

**Reply by tomyems** [reply]
We can achieve AA compliance by just shifting the red component from 212 to 205 (that is from d4351c to cd351c). This is very low risk. There's some concern about the darker red not being perceptible in low contrast environments.

The change is (almost) imperceptible :)

![image](https://user-images.githubusercontent.com/93599300/210392609-1a7406cb-ebc8-4c1d-955f-f1f9509267d5.png)

**Reply by tomyems** [reply]
Deadline for response to changing the red to cd351c is 19 January.

**Reply by FlorrieA** [reply]
Hi there! On the new red, would cd351c be the new red for all instances going forward where red is used in Home Office services, or just in instances where it is used on error messages?? Thank you

**Reply by tomyems** [reply]
Good question, where were you expecting to use it? CD351C achieved 4.5 contrast with the light grey background, which was the reason behind changing it.

The other reds in the design system say for alerts and expandable banners are different anyway. They also have a different context, as they are backgrounds with text on top of it, and will be reviewed separately.

**Reply by FlorrieA** [reply]
Thanks @tomyems. It's for a banner and an alert so it sounds like moving to CD351C on these perhaps isn't appropriate at this point. I'll use the GOV.UK red (and any other shades in GOV.UK Design System) where text is on top of it or we have a white background but use CD351C for error messages on a grey background in that case, but let me know if you'd suggest otherwise.

**Comment by Richard592** [comment]
So, was there push back on A70000? I'm happy with the very low risk move to CD351C if it improves contrast. Go go go.

**Reply by tomyems** [reply]
I believe CD351C is lower risk than A70000. There's less change to the colour. I was a bit worried about making the green and blue components 0. Keeping the green and blue components also means there's less change to the red component to achieve sufficient contrast.

**Reply by andymarshall-35** [reply]
Appreciate I'm over a year late to the party, but raising this issue again, in case the newer APCA guidelines has any place to revisit this one (see below comment - https://github.com/UKHomeOffice/design-system/discussions/355#discussioncomment-9264126 )

**Comment by larab-codes** [comment]
### Findings related to the use of tinted backgrounds in an enterprise system. 

## About the system
The system supports a range of users all with different requirements, many of them are on it for the majority of their working day. 
The system is not based on either the Home Office or the Gov.uk design system, however, it is slowly moving towards the Gov.uk one.
Ultimately, that means the interface background is pure white, with black text. Many users report that they suffer eye strain as a result. Dyslexic users are worse affected by the stark black on white text as it causes the text to 'dance', making it even harder to read and digest it. It is estimated that as much as 10% of the general population experience some degree of dyslexia. 

Users also struggled with understanding the page hierarchy. As part of their job, they need to consider and view vast amounts of information, often across multiple systems. To help with this we introduced 'overview' screens that collated information from different areas of the system onto one screen. We provided several ways to customise their view, accordions, filters and content areas bounded by background tints - all of which tested well in usability testing. 

## Background tints
 The addition of the background tints helped by

- Grouping related information
- Making scanning the page easier
- Reinforcing with accordions were open (the tinted background content was revealed when they were open) and which were not
- Softening the stark black and white 'dancing text' issue for dyslexic users
- Addressing some of the eye strain issues

![image](https://user-images.githubusercontent.com/68660397/211523238-fe4fc76b-3167-4406-a4d0-711f3a2bbc05.png)


The proposed design shows part of the screen, the filters are shown, and the first accordion is closed but the subsequent two are open. The revealed content is grouped with tinted backgrounds. Most content is blue, but on certain complex summaries, there is also a yellow section showing further information.  



# Response   
The overall response to the proposed design was positive in usability research, with users saying that it made their job easier. We tested with users a range of users, some of whom said they had dyslexia. 
Dyslexic users also told us that a tinted background of any colour helped them with the 'dancing text' issue. But it's worth noting that a lot of dyslexic users choose peach as their preferred background colour. The colours proposed in our design worked better with the existing colour scheme and were also effective. 
It is however best practice to allow users to customise their own technology, so try to make sure your solution is compatible with the tools your users have access to. In our case, the system did not allow them to customise their view without complicated workarounds.

**Comment by tomyems** [comment]
### Purple
The Home Office has updated it's purple to #723382, so the design system should follow suit.

**Reply by tomyems** [reply]
P.S. there is an internal DDaT purple which is currently #8F23B3 (the current one published) we believe that this has remained and not been updated when the Home Office purple was updated to #723382 in 2021.

**Comment by andymarshall-35** [comment]
Is there any merit in reviewing the Home Office colour palette against the APCA guidelines?
I appreciate the need to pass WCAG2.2 AA / AAA compliance, but as I understand it, APCA offers a more realistic approach to colour contrast / readability (plus, future proofs us for WCAG3).
It also offers more granular success criteria for contrast ratios based on the text size and weight, as well as picking up on some false positives that the standard contrast ratios miss.
(It also distinguishes between foreground vs background colours (ie dark on light is not the same as light on dark) not presenting the same, but I don't imagine this is so important as we don't have an implemented dark mode)

For anyone who's not familiar with APCA, start here  - https://ruitina.com/apca-accessible-colour-contrast/
More info / testing tools here - https://www.myndex.com/APCA/ 
Clearer testing tool here - https://www.achecks.org/apca-wcag-3-accessible-colour-contrast-checker/

Using the above example of the revised red error color: 
.govuk-body (19px on small and large screens, and 400 weight) for the proposed revised red used for errors (#CD351C) on the HO light grey, doesn't pass:
<img width="719" alt="Screenshot 2024-04-29 at 15 50 21" src="https://github.com/UKHomeOffice/design-system/assets/167748704/64e0e1c8-75ba-483b-b125-5a7b83646ae6">

whereas the alternative #A70000 does pass at that size and weight:
<img width="686" alt="Screenshot 2024-04-29 at 15 53 08" src="https://github.com/UKHomeOffice/design-system/assets/167748704/1d116dd2-d197-4e12-ae9c-132f5fcbd8b1">

Unless theres a good reason we're not adopting APCA at this stage of course - still very new here!

**Reply by tomyems** [reply]
We haven't actually implemented the change yet https://github.com/UKHomeOffice/design-system/issues/489 - to be honest this was a quick fix, to reduce friction with testing to at least meet AA. The colour palette needs some closer attention and more detailed work.

**Reply by andymarshall-35** [reply]
Fair enough - a quick fix is still better than nothing!

The APCA tools aren't that easy to use, and not widely adopted in plugins / extensions yet, so running the colour palette thru this won't be a quick task, but definitely worthwhile if / when we do have a proper review of the colour palette.
I've also been looking at the X-Gov Dark Mode topic (https://github.com/alphagov/govuk-design-system-backlog/issues/177) that may or may not become relevant to this task, as the whole dark text on light bg vs light text on dark bg can be tested too.

**Comment by mollyhayne1** [comment]
**Styling of GOV.UK tabs**

Component: [https://design-system.service.gov.uk/components/tabs/](https://design-system.service.gov.uk/components/tabs/)

Do we have a colour palate we can apply to tabs used in an internal system? I've noticed the styling of the GOV.UK tabs component is consistent with HO styling when using the #F5F5F5 background. There is a low contrast with the inactive tabs, and a white background on the active tab.

Example:
![Tabs GOV UK styling](https://github.com/user-attachments/assets/37771479-b6dc-4e71-b2d0-e3f42487866f)


I’ve seen some internal services using this GOV.UK styling, but others have used override classes to adapt it. On our internal caseworking service we have used override classes to: 
- remove the white background  
- make the background color of inactive tabs darker (#cbcbcb) 

Example: 
![Tabs alternative styling](https://github.com/user-attachments/assets/bd732c42-62df-4b2b-b25d-16d76cdfa1c8)

We want to make sure our designs are consistent with other internal services across the Home Office. Do we have any standard override classes / styling that we can apply to ensure this?
